base/replutil.jl

```diff
@@ -106,11 +106,14 @@ function showerror(io::IO, ex::DomainError, bt; backtrace=true)
         code = ccall(:jl_lookup_code_address, Any, (Ptr{Void}, Cint), b-1, true)
         if length(code) == 5 && !code[4]  # code[4] == fromC
             if code[1] in (:log, :log2, :log10, :sqrt) # TODO add :besselj, :besseli, :bessely, :besselk
-                println(io,"\n$(code[1]) will only return a complex result if called with a complex argument.")
-                print(io, "try $(code[1]) (complex(x))")
-            elseif (code[1] == :^ && code[2] == symbol("intfuncs.jl")) || code[1] == :power_by_squaring
-                println(io, "\nCannot raise an integer x to a negative power -n. Make x a float by adding")
-                print(io, "a zero decimal (e.g. 2.0^-n instead of 2^-n), or write 1/x^n, float(x)^-n, or (x//1)^-n")
+                println(io,"\n$(code[1]) will only return a complex result if called with a complex argument. ",
```

Comment:
@stevengj , actually that was the original design. I didn't *introduce* it. In fact, I'm not getting *extraneous newlines* comparing the output before and after your commit removing those `println` (https://github.com/JuliaLang/julia/commit/e3d2cb769f2ae24d09db3c0694ebedcfadd80224). Since I didn't notice that behavior, I didn't bother to change the old code.

**Before**
```julia
 _/ |\__'_|_|_|\__'_|  |  Commit 732860a (0 days old master)
|__/                   |  x86_64-apple-darwin14.5.0

julia> log(-2)
ERROR: DomainError:
log will only return a complex result if called with a complex argument.
try log (complex(x))
 in log at math.jl:139

julia> (-2.0)^(2.2)
ERROR: DomainError:
 in ^ at math.jl:275
```

**After**
```julia
 _/ |\__'_|_|_|\__'_|  |  Commit e3d2cb7 (0 days old master)
|__/                   |  x86_64-apple-darwin14.5.0

julia> log(-2)
ERROR: DomainError:
log will only return a complex result if called with a complex argument. Try log (complex(x)).
 in log at math.jl:139

julia> (-2.0)^(2.2)
ERROR: DomainError:
Exponentiation yielding a complex result requires a complex argument.
Replace x^y with (x+0im)^y, Complex(x)^y, or similar.
 in ^ at math.jl:275
```